projects/01-rubygems-cli/solution/lib/program.rb

```diff
@@ -0,0 +1,23 @@
+require "./lib/show_command"
+require "./lib/search_command"
+require "./lib/help_command"
+require "./lib/command_results/command_error_result"
+
+class Program
+  class << self
+    def execute(argv)
+      return CommandErrorResult.new("No command provided!", 1) if argv.nil? || argv[0].nil? || argv[0].empty?
```

Comment:
`argv` should always be an array, so don't think it makes sense to check if it's `nil`.
```ruby
argv.empty? # this should suffice
```

Reply:
96261d7af86a9c2fcd730b78a9ac12544657f6c9